### PR TITLE
[span processor] Add force_flush method.

### LINKED ISSUE #349: Implement SpanProcessor::force_flush
Implement `force_flush` as specified here: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#forceflush

> Exports all spans that have not yet been exported to the configured Exporter.
> 
> ForceFlush SHOULD provide a way to let the caller know whether it succeeded, failed or timed out.
> 
> ForceFlush SHOULD only be called in cases where it is absolutely necessary, such as when using some FaaS providers that may suspend the process after an invocation, but before the Processor exports the completed spans.
> 
> ForceFlush SHOULD complete or abort within some timeout. ForceFlush can be implemented as a blocking API or an asynchronous API which notifies the caller via a callback or an event. Language library authors can decide if they want to make the flush timeout configurable.

### PR DESCRIPTION
**Pull request by TommyCpp**
See https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#forceflush for more details.

closes #349

### CODE REVIEW

**Review by frigus02 — COMMENTED** [review]
This looks nice. I'm unsure about this part of the spec:

> ForceFlush SHOULD provide a way to let the caller know whether it succeeded, failed or timed out.

This sounds like, similar to shutdown, the function should wait until the spans have been exported and only return then.

I don't immediately know how to implement this, though. And it's only a "should". So I think we're spec compliant with the way you did it.

**Review by flycash — COMMENTED** [review]
From my understanding,
```
ForceFlush SHOULD provide a way to let the caller know whether it succeeded, failed or timed out.
...
ForceFlush SHOULD complete or abort within some timeout. 
```
I think `force_flush` need to add the `timeout` parameter, and use enum or status code as return value.

Implementations can use something like `future` to flush spans.

**Comment by TommyCpp** [comment]
> ForceFlush SHOULD provide a way to let the caller know whether it succeeded, failed or timed out.

@frigus02 Yeah I also noticed that, but I find the `shutdown` method also doesn't return any result. So feel like we should have a dedicated PR to discuss how to implement this. 

> I think force_flush need to add the timeout parameter, and use enum or status code as return value.

@flycash I agree. But I think we might as well add timeout for both `shutdown` and `force_flush` in `SimpleSpanProcessor` and `BatchSpanProcessor`. So I was going to create another PR for that.

File: opentelemetry/src/sdk/trace/span_processor.rs [hunk 1]
```diff
@@ -500,4 +516,35 @@ mod tests {
         assert_eq!(builder.config.max_export_batch_size, 120);
         assert_eq!(builder.config.max_queue_size, 120);
     }
+
+    #[tokio::test]
+    async fn test_batch_span_processor() {
+        let (exporter, mut export_receiver, _shutdown_receiver) = new_tokio_test_exporter();
+        let mut config = BatchConfig::default();
+        config.scheduled_delay = Duration::from_secs(60 * 60 * 24); // set the tick to 24 hours so we know the span must be exported vis force_flush
```
Comment (frigus02) [inline opentelemetry/src/sdk/trace/span_processor.rs:524]:
```suggestion
        config.scheduled_delay = Duration::from_secs(60 * 60 * 24); // set the tick to 24 hours so we know the span must be exported via force_flush
```

**Review by frigus02 — APPROVED** [review]
Makes sense to me 👍

**Pull request merged**